### PR TITLE
feat(map): add direct location editing with shared-location safeguards

### LINKED ISSUE #734: Add direct location editing in Map
Add direct location editing in Map. Today AO and location are effectively combined in the edit flow, but location is its own data object and may be shared across multiple AOs. We should support editing location directly while making it clear which AOs/events will be affected. If a location is shared, the UI should warn the user and show the linked AOs/events. We should also support the case where a user wants to move a single AO off a shared location without unintentionally changing the others.

## Problem
- In Map, AO and location are currently bundled, so there is no clear way to edit the location by itself.
- That gets risky when multiple AOs share one location, because changing the location could unintentionally affect more than one AO.

## Goal
- Let users edit a location directly from Map.
- Make the impact clear when that location is shared by multiple AOs/events.

## Proposed behavior
- Add separate actions for:
  - Edit AO
  - Edit Location
- In Edit Location, show which AOs/events are tied to that location before saving.
- Warn that editing the location will affect all linked AOs using that same location.
- If a user wants to split one AO off from a shared location, support giving that AO its own location instead of silently changing the shared one.

## Edge cases to test
- Two AOs sharing one location
- One AO needing to move off a shared location
- How shared locations render in Map after edits
- Whether events/AOs linked to the location still display correctly after changes

## Acceptance criteria
- Users can edit a location directly in Map
- Users can see all affected AOs/events before saving
- Shared-location edits do not feel ambiguous
- Users have a path to separate an AO from a shared location when needed

### PR DESCRIPTION
**Pull request by dnishiyama**
Closes #734.

## Problem
In Map, AO and location are edited together (`edit_ao_and_location`), so there's no way to edit a location by itself. Because one location row can be **shared by multiple AOs** (events from different AOs pointing at the same `locationId`), editing it today silently changes it for every AO that uses it.

## What this does
Splits the flow into two dedicated request types and surfaces the blast radius before saving.

- **New request types `edit_ao` and `edit_location`** (the existing `edit_ao_and_location` is kept for in-flight/pending requests). Includes the `request_type` enum migration, validator schemas, apply handlers, submit + admin-approval procedures, and map-event mapping.
- **New `location.linkedAos` query** — returns the distinct active AOs and events at a location (private events counted but not detailed) to drive the warning UI.
- **Map UI** — separate **"Edit AO details"** and **"Edit location details"** actions. The new Edit Location modal lists every AO/event at the location and, when it's shared, **requires an explicit acknowledgment** before saving and offers a per-AO **"move to its own location"** escape hatch (reusing the existing `move_ao_to_different_location` flow). Marker-drag toast now points at "Edit location details".
- **Admin** — dedicated EditAo / EditLocation review forms; the location form warns reviewers when the location is shared.

## Acceptance criteria (from #734)
- ✅ Users can edit a location directly in Map
- ✅ Users can see all affected AOs/events before saving (`linkedAos` + notice)
- ✅ Shared-location edits are unambiguous (warning + required acknowledgment)
- ✅ Path to separate an AO from a shared location (per-AO "move to its own location")

## QA done
- Full migration chain incl. `0018` applies cleanly on **real Postgres 17.6**; `request_type` enum gains `edit_ao`/`edit_location` in the correct position.
- Integration-checked on real PG: `linkedAos` aggregation (shared location, private-event counting) and that a location edit leaves all AOs' events intact (none reassigned).
- Typecheck (23/23 packages), ESLint, Prettier, commitlint all green.
- Tests: validator schemas, `open-request-modal` mapping/guards, modal submission wiring (map suite 156 pass), and handler unit tests asserting `edit_location` never touches an AO and `edit_ao` never touches a location (api suite 16 pass).

## Still pending
- **Live UI walkthrough** — couldn't run the app stack locally (no access to F3's `f3-authentication-staging` secrets). Intended to verify on this PR's **preview deploy**: the shared-location warning + acknowledgment gate, the "move AO to its own location" split, and the admin review forms.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added separate workflows for editing AO and location details.
  * Added dedicated forms and modals with validation and submission support.
  * Added warnings and acknowledgment requirements for locations shared by multiple AOs.
  * Added visibility into linked AOs and their active events.
  * Admin review now supports approving AO-only and location-only edits.

* **Bug Fixes**
  * Updated edit guidance to direct users to the correct location-editing option.

* **Tests**
  * Expanded coverage for validation, modal behavior, submissions, and update processing.

### CODE REVIEW

**Comment by dnishiyama** [comment]
@coderabbitai full review

File: apps/admin/src/app/_components/forms/edit-ao-request-form.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  const originalAoId = meta?.originalAoId;
+
+  const { data: orgResponse } = useQuery(
+    orpc.org.byId.queryOptions({
+      input: { id: Number(originalAoId) },
+      enabled: originalAoId != null,
+    }),
+  );
+  const org = orgResponse?.org;
```
[the remark this answers is not in the document]
Comment (dnishiyama) [reply apps/admin/src/app/_components/forms/edit-ao-request-form.tsx:25]: Backstopped server-side instead of gating the admin UI on query-loading state: `handleEditLocation` now rejects an unacknowledged edit to a location shared by >1 active AO regardless of which path it comes through (direct submit or admin approval) — see the reply on the server-side-enforcement thread below. Admin approval specifically sets `acknowledgeShared ??= true`, on the reasoning that reaching the approval path means a reviewer used the `EditLocationRequestForm`, which already fetches and displays the linked-AOs blast-radius warning before they click Approve — the review itself is the acknowledgment. Didn't additionally gate the Approve button on the form's query state, since the server enforces the actual constraint unconditionally now; that would be added UI complexity for a case the server already can't be bypassed on.

File: apps/map/src/app/_components/forms/linked-aos-notice.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      enabled: locationId != null,
+    }),
+    throwOnError: false,
+  });
+
+  const isShared = (data?.totalAoCount ?? 0) > 1;
+
+  useEffect(() => {
+    onSharedChange?.(isShared);
+  }, [isShared, onSharedChange]);
```
[the remark this answers is not in the document]
Comment (dnishiyama) [reply apps/map/src/app/_components/forms/linked-aos-notice.tsx]: Fixed — `LinkedAosNotice` now only reports a shared/not-shared verdict once the query has resolved successfully (`resolved = locationId == null || (!isLoading && !isError && !!data)`), shows a retryable error state on failure, and `EditLocationModal` tracks `sharedCheckResolved` separately from `isShared` so submission stays disabled until a successful check completes.

File: apps/map/src/app/_components/forms/linked-aos-notice.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+}) => {
+  const { data, isLoading } = useQuery({
+    ...orpc.location.linkedAos.queryOptions({
+      input: { locationId: locationId ?? 0 },
+      enabled: locationId != null,
+    }),
+    throwOnError: false,
+  });
+
+  const isShared = (data?.totalAoCount ?? 0) > 1;
```
[the remark this answers is not in the document]
Comment (dnishiyama) [reply apps/map/src/app/_components/forms/linked-aos-notice.tsx:48]: Deliberate active-only scope, not changed. The `linkedAos` endpoint counts AOs with currently-active events at this location — the actual, present blast radius. Warning about AOs that are dormant now but *might* be reactivated later is a real but speculative edge case (would need to predict future reactivation, not describe current state), and is a product-policy call beyond a review-fix pass. Happy to revisit if this becomes a real incident.

File: apps/map/src/app/_components/modal/update/edit-location-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  data: DataType[ModalType.EDIT_LOCATION];
+}) => {
+  const form = useForm({
+    schema: EditLocationSchema,
+    defaultValues: data,
+    mode: "onBlur",
+  });
+
+  const [isShared, setIsShared] = useState(false);
+  const [acknowledged, setAcknowledged] = useState(false);
```
[the remark this answers is not in the document]
Comment (dnishiyama) [reply apps/map/src/app/_components/modal/update/edit-location-modal.tsx]: Fixed — applied essentially your proposed diff: added `sharedCheckResolved` state, `onSharedChange` sets both `isShared` and `sharedCheckResolved` together (only called once `LinkedAosNotice` resolves), and `disabled={!sharedCheckResolved || (isShared && !acknowledged)}`.

File: packages/api/src/__test__/fixtures/update-request.ts [hunk 5]
```diff
@@ -112,6 +114,46 @@ export const createEditAOAndLocationRequest = (
...
+  originalRegionId: 1,
+  originalLocationId: 1,
+  locationLat: 36.1,
+  locationLng: -81.2,
+  locationAddress: "999 Updated Ave",
+  locationCity: "Boone",
+  locationState: "NC",
+  locationZip: "28607",
+  locationCountry: "United States",
+  currentValues: {},
```
[the remark this answers is not in the document]
Comment (dnishiyama) [reply packages/api/src/__test__/fixtures/update-request.ts:153]: Not changed — `"test-request-id"` is the existing convention across all 10 fixture helpers in this file, and none of them are ever passed through `BaseSchema.parse()`/`.safeParse()` in the test suite (checked — they're passed directly into handler functions like `handleEditLocation(ctx, createEditLocationRequest())`), so the UUID constraint never actually runs against them. Changing just this one fixture to a real UUID would make it the odd one out for no test-behavior benefit.

File: packages/api/src/lib/update-request-handlers.ts [hunk 6]
```diff
@@ -458,6 +460,34 @@ export const handleEditAOAndLocation = async (
...
+    locationLng: request.locationLng,
+    locationAddress: request.locationAddress,
+    locationAddress2: request.locationAddress2,
+    locationCity: request.locationCity,
+    locationState: request.locationState,
+    locationZip: request.locationZip,
+    locationCountry: request.locationCountry,
+    locationDescription: request.locationDescription,
+  });
+};
```
[the remark this answers is not in the document]
Comment (dnishiyama) [reply packages/api/src/lib/update-request-handlers.ts:523]: Fixed — this was the real gap (client-only disabled button, bypassable via direct API call). `EditLocationSchema` gained `acknowledgeShared: z.boolean().optional()`; `handleEditLocation` now counts distinct active-AO `orgId`s sharing `request.originalLocationId` and throws `ORPCError("CONFLICT")` if there's more than one and `acknowledgeShared` isn't set. The map modal sends `acknowledgeShared: acknowledged` on submit. Added handler tests for both the reject and the acknowledged-allow paths.

**Comment by dnishiyama** [comment]
@coderabbitai review

**Comment by dnishiyama** [comment]
Adversarial review round 2 — responses (commit `26b3ea5f`):

1. **Admin approval auto-acknowledges, bypassing the shared-location gate** — 🚩 flagged `[A+B]` (both reviewers independently). Fixed per Declan's call: `EditLocationRequestForm` now blocks Approve (new `onApprovalGateChange` prop) until the linked-AO check resolves, and — if shared — until the reviewer explicitly checks the same acknowledgment checkbox the map UI already has. Removed the blanket `acknowledgeShared ??= true` auto-default in `normalizeAdminRequestInput`; the reviewer's real checkbox state now flows through the shared form into the same server-side `handleEditLocation` guard. (Reviewer A's citation of `map-update-request-flow.md §5` doesn't actually exist in that section — checked directly — but the underlying technical observation was accurate and is now fixed regardless.)
2. **linkedAos endpoint has no response bound** — fixed defensively. Realistic F3 locations are shared by a handful of AOs, so this wasn't a live risk, but added a generous 2000-row cap as a guard against a pathological/corrupted-data location.
3. **linkedAos auth tier (protectedProcedure) vs spec** — verified against `specs/map-browse-and-search.md`: matches the documented model exactly ("map API read procedures are protectedProcedure, not publicProcedure... trusted caller"). Not a deviation, not changed.
4. **handleEditLocation loads all events into memory** — fixed. Replaced with a single `COUNT(DISTINCT orgId)` aggregate query.
5. **LinkedAosNotice doesn't guard nullable dayOfWeek/startTime** — the crash claim didn't hold (`getShortDayOfWeek`/`formatTime` already handle null → `""`), but a location with both fields null was rendering dangling empty parens — fixed to omit the parenthetical when there's no schedule to show. Added a test file for this component (previously untested) covering the fail-closed states and the null-schedule case.

Verified: full `pnpm test` on f3-api (27/27) and f3-map (166/166, up from 161), typecheck/lint clean on f3-api/f3-map/f3-admin.

File: apps/map/src/app/_components/forms/linked-aos-notice.tsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  // Fail closed: only report a shared/not-shared verdict once the check has
+  // actually resolved. While loading or after an error we deliberately stay
+  // silent so the parent keeps submission gated rather than defaulting to
+  // "not shared" and letting an unacknowledged shared edit through.
+  const resolved = locationId == null || (!isLoading && !isError && !!data);
+  const isShared = (data?.totalAoCount ?? 0) > 1;
+
+  useEffect(() => {
+    if (resolved) onSharedChange?.(isShared);
+  }, [resolved, isShared, onSharedChange]);
```
[the remark this answers is not in the document]
Comment (dnishiyama) [reply apps/map/src/app/_components/forms/linked-aos-notice.tsx:52]: Already fixed — this finding is stale against current code. `linked-aos-notice.tsx`'s `resolved` is `!isLoading && !isError && !!data` and is reported unconditionally on every change (not just when it becomes true), so a background refetch failure flips `isError` back to true and re-closes the gate; `edit-location-modal.tsx`'s `SubmitSection` is `disabled={!sharedCheck.resolved || ...}`. There's already a regression test for exactly this scenario: `linked-aos-notice.test.tsx`'s "re-closes the gate when a refetch fails after an earlier successful resolution" (passing). No `placeholderData`/`keepPreviousData` is used on this query, so `isError` genuinely does flip on a failed refetch rather than being masked by stale cached data.
